recipe-box-frontend/src/main/frontend/src/components/RecipesList/RecipesGrid.tsx

```diff
@@ -0,0 +1,184 @@
+import { useEffect, useState } from "react";
+import RecipeCard from "./RecipeCard";
+import "./RecipesGrid.css";
+
+interface Recipe {
+    id: number;
+    name: string;
+    description: string;
+    imageUrl: string;
+    mealType: string;
+}
+const backendUrl = process.env.REACT_APP_BACKEND_URL; //  http://localhost:8080/recipes/
+export default function RecipesGrid({
+    mealType,
+    searchQuery,
+}: {
+    mealType: string;
+    searchQuery: string;
+}) {
+    const [recipes, setRecipes] = useState<Recipe[]>([]);
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(() => {
+        const controller = new AbortController();
+        const signal = controller.signal;
+
+        let url = `${backendUrl}${mealType ? `mealType=${mealType}` : ""}`;
+
+        fetch(searchQuery ? `${url}/search?recipeName=${searchQuery}` : url, {
+            signal,
+        })
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error(`Bad response: ${response.status}`);
+                }
+                return response.json();
+            })
+            .then((data: Recipe[]) => {
+                if (!signal.aborted) {
+                    setRecipes(data);
+                }
+            })
+            .catch((error) => {
+                if (error.name !== "AbortError") {
+                    console.error("Fetch error:", error);
+                    setError("Fetching failed");
+                }
+            });
+
+        return () => {
+            controller.abort();
+        };
+    }, [mealType, searchQuery]);
+
+    if (error) {
```

Comment:
We want to render the mocked data in case the fetched object is empty, not when there is an error. 
For a cleaner code, I suggest to create a variable for all recipes, where we use recipes if the fetched data is not empty, otherwise we should use the testRecipes. Then, we can use this new variable to filter based on mealType and searchQuery. If we do this, then we will have only one return where we map the recipes.